src/students/students.controller.spec.ts

```diff
@@ -26,7 +29,48 @@ describe('StudentsController', () => {
 
   it('should be defined', () => {
     expect(controller).toBeDefined();
-    controller.findAll(requestMock, statusResponse);
-    expect(statusResponse.status).toHaveBeenCalledWith(200);
+  });
+
+  it('should create new student', async () => {
+    const response = await controller.create(mockStudent);
+    expect(response.status).toEqual('success');
+    expect(response.data).toBeDefined();
+    expect(response.data).toEqual({ id: expect.any(Number), ...mockStudent });
+
+    mockStudentId = (response.data as Student).id;
+  });
+
+  it('should find all students', async () => {
+    const response = await controller.findAll();
+    expect(response).toBeDefined();
+    expect(Array.isArray(response.data)).toBe(true);
+    expect((response.data as Student[]).length).toBeGreaterThan(0);
+  });
+
+  it('should find one student', async () => {
+    const response = await controller.findOne(mockStudentId.toString());
+    expect(response).toBeDefined();
+    expect((response.data as Student).name).toEqual(mockStudent.name);
+  });
```

Comment:
Make separate methods for separate test cases. (Also applicable for rest of the files)